### PR TITLE
Improve cody history and add CTAs

### PR DESCRIPTION
**Pull request by thenamankumar**
- Improve Cody History UI on the web based on [Figma](https://www.figma.com/file/p10j2PX1cpZ374Vrc9ibuc/Cody-chat%E2%80%A8-history-view?type=design&node-id=2-461&t=6NPraRq4p1F1LlEi-0)
- Replace Cody logo with a colorful variant
- Solve chat history bug on dotcom caused due to different schema of old chats already stored in localstorage.
- Add VS Code Extension ads.

image
image

image

image
image

## Test plan

- cody web

### CODE REVIEW

**Comment by sourcegraph-buildkite** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.01% (+0.34 kb) | 0.01% (+1.33 kb) | 0.01% (+0.99 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/naman/cody-history-upgrade/compare-report.html;tab=live_object) for a full comparison between the commits f38d91f and 4feda61 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

**Comment by almeidapaulooliveira** [comment]
Screenshot 2023-05-09 at 7 18 39 PM

@thenamankumar, what action does this icon take? My assumption is that it will switch back to the selected conversation after clicking that button, but I'm not sure.

I was expecting that when users switched from the chat to the list of chats it would unselect the current chat. Then users would select another chat to work on (much like WhatsApp on mobile).

However, I am open to understanding your perspective. Please let me know.

**Comment by almeidapaulooliveira** [comment]
A few minor fixes:
- We're [standardizing](https://docs.google.com/document/d/1YYBvCA2tcvudKMJjjPf7kki6e_eCeWJT93AeCCH4WRw/edit#bookmark=id.uhgxoygqqjr5) the wording as 'chats' to be more friendly for users. So the tooltips would be:
  - 'Delete all chats' for deleting all conversations icon.
- We're changing the [history paradigm ](https://docs.google.com/document/d/1YYBvCA2tcvudKMJjjPf7kki6e_eCeWJT93AeCCH4WRw/edit#bookmark=id.e75flzw2iyyx)to active conversations. So the tooltip would be:
  - 'Active chats' for the switch view button.

**Comment by almeidapaulooliveira** [comment]
I ran locally and the delete button appeared unaligned, but the PR screenshots look fine. Not sure if it's an issue.

Screenshot 2023-05-09 at 7 47 55 PM

File: client/web/src/cody/chat/AskCodyThumbnail.tsx [hunk 1]
```diff
@@ -0,0 +1,186 @@
+import React, { SVGProps } from 'react'
+
+export const AskCodyThumbnail: React.FunctionComponent<React.PropsWithChildren<SVGProps<SVGSVGElement>>> = (
+    props: SVGProps<SVGSVGElement>
+) => (
+    <svg
```
Comment (philipp-spiess) [inline client/web/src/cody/chat/AskCodyThumbnail.tsx:6]: There's a lot going on in this SVG, did you pipe this throughs svgo yet to minimize it yet? It seems like a straight export (it even contains global IDs and such).

File: client/web/src/cody/chat/CodyChatPage.tsx [hunk 2]
```diff
@@ -39,25 +32,31 @@ import { ChatUI } from '../components/ChatUI'
...
 export const CodyChatPage: React.FunctionComponent<CodyChatPageProps> = ({ authenticatedUser }) => {
     const { reset, clearHistory } = useChatStore({ codebase: '' })
     const [enabled] = useFeatureFlag('cody-web-chat')
-    const onInstallClick = (): void => eventLogger.log(EventName.TRY_CODY_VSCODE)
-    const onMarketplaceClick = (): void => eventLogger.log(EventName.TRY_CODY_MARKETPLACE)
+    const [showVSCodeCTA, ____] = useState<boolean>(Math.random() < 0.5)
+    const [isCTADismissed = true, setIsCTADismissed] = useTemporarySetting('cody.chatPageCta.dismissed', false)
+    const onCTADismiss = (): void => setIsCTADismissed(true)
+    const onDownloadVSCodeClick = (): void => eventLogger.log(EventName.CODY_CHAT_DOWNLOAD_VSCODE)
+    const onTryOnPublicCodeClick = (): void => eventLogger.log(EventName.CODY_CHAT_TRY_ON_PUBLIC_CODE)
```
Comment (philipp-spiess) [inline client/web/src/cody/chat/CodyChatPage.tsx:53]: Nit: looks like you can move this outside of the function scope into the module scope

File: client/web/src/cody/chat/CodyChatPage.tsx [hunk 3]
```diff
@@ -39,25 +32,31 @@ import { ChatUI } from '../components/ChatUI'
...
 
+const VsCodeExtensionThumbnail = lazyComponent(() => import('./VsCodeExtensionThumbnail'), 'VsCodeExtensionThumbnail')
+const AskCodyThumbnail = lazyComponent(() => import('./AskCodyThumbnail'), 'AskCodyThumbnail')
+
 export const CodyChatPage: React.FunctionComponent<CodyChatPageProps> = ({ authenticatedUser }) => {
     const { reset, clearHistory } = useChatStore({ codebase: '' })
     const [enabled] = useFeatureFlag('cody-web-chat')
-    const onInstallClick = (): void => eventLogger.log(EventName.TRY_CODY_VSCODE)
-    const onMarketplaceClick = (): void => eventLogger.log(EventName.TRY_CODY_MARKETPLACE)
+    const [showVSCodeCTA, ____] = useState<boolean>(Math.random() < 0.5)
```
Comment (philipp-spiess) [inline client/web/src/cody/chat/CodyChatPage.tsx:49]: You can just ignore the second value I think!

```suggestion
    const [showVSCodeCTA] = useState<boolean>(Math.random() < 0.5)
```

Comment (philipp-spiess) [inline client/web/src/cody/chat/CodyChatPage.tsx:49]: You can just ignore the second value I think!

```suggestion
    const [showVSCodeCTA] = useState<boolean>(Math.random() < 0.5)
```

File: client/web/src/cody/chat/AskCodyThumbnail.tsx [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
...SKhE4kAZsJZixv7B4KVxK5U/sG648qnmcnq40RWjyZFtKZ4jkjJ1Zq0eDxN408lVbMvKZP5tfN0mpoT92VlbgZHa50imdXVhmps7riQlNWFpTqPp+l9AvRjOXbwxeiZz+rPDRPWdjr1tGGepAUBQUAQEAQEAUGgdQiciY1DKf0exuHi6gRvL50AoHW9nH/tg4eO4rXXP8R/P30LPj7tvF8HP1LHmxpJNEAgN/wk7DoGwMLeHsXJyShNT1N1zGivDKeuXRvUvxAZS35bCQ93N4waNfRCdHdZ9XE6+gx86fq2tb34G1sfOXoCBw4cwf1z7risMJDJCAKCwNWFAFtdVtHT2OLhfHW977JaQeCyQWD3oRhlpfHArNGwJQKUw8HeGlk5hfhz1yk8cudYDOgZiAoynHewqyVMm1pAfXK5/nlT7a7EfCZ1tXC0sUAFEcwcmaRg/mZbBF5fdQSfbjypyGRyptCHKX3xM9nM4XjuqLVl4tfPxU5f18+1Nq3PbCLBBPfKI/F484+j6sUWG1lFDRXnGplt2M1NAzrRhoSFeG/tMbxPBDgrsDlqqqtQUVbc6KuG/0iUEAQEAUFAEBAEBIE2IbB27Z/4cf5i/OvVd7Bs2Zo29XU+jR0dHODn5w0T2iS6vaIoMQFH33gNK4cOwsohA1W6ury8zvCZ+/ch59ixOnmX8iTlz0048MLzdaZw4v13Ebt4UZ28C3Wy+7FHkX3ksOoufc8uhH/2KQ6+8hL2Pff0hRqiTj984+O/n3+Hbt1C6+RfzidJSSnYtXu/UVP857/eQXh4lFF1jan05+YdyMzMarRql9DOWP3HRuzdd6jRcskUBAQBQaC9EDAj4ZkonNsLbRlHEBAE9AhEx6Vj8ar9uHFyf4Ses9PgQk83R1Xn8XuuQZC/OwqItCyvqNLn6ztoImGoZjZMN1H9L5e9NSIFNqQQf2laH5ibmSoS19hFWpqbIjW/hOwtdERzal6JUicb096WxuRNBdmTublwoxsK9aNjB3s8O7kX2XmUY8PJRPy8R2fHUZibhl2rPq9fXZ2LpUajsEimICAICAKCgCDQKgQeefheVf+LL79vVbsLVTkkJAiffvzmherOqH72Pfs0KXcdMPyreagsKsb+F56FpbMzuj3ymL59wqqVsHJ3h0uvXvq8S5koz89H/hndfifaPEpSU2FqZaWdXrRj4I03g
```
Comment (philipp-spiess) [inline client/web/src/cody/chat/AskCodyThumbnail.tsx:183]: Oh this contians an image? Should we rasterize it instead and handle it as an image?

File: client/web/src/cody/chat/VsCodeExtensionThumbnail.tsx [hunk 5]
```diff
@@ -0,0 +1,294 @@
+import React, { SVGProps } from 'react'
```
Comment (philipp-spiess) [inline client/web/src/cody/chat/VsCodeExtensionThumbnail.tsx:1]: This SVG is also quite massive. Have you tried piping it through svgo? 🤔

**Review by philipp-spiess — APPROVED** [review]
Looks good from a code perspective, the SVGs are huge though. I think we need to see if we can:

- Use svgo to minimize them (and avoid clashing global IDs)
- Compare how much the `png` would weigh and consider that instead?

File: client/web/src/cody/chat/CodyChatPage.tsx (hunk 3 above)
Comment (thenamankumar) [reply client/web/src/cody/chat/CodyChatPage.tsx]: done

Comment (thenamankumar) [reply client/web/src/cody/chat/CodyChatPage.tsx]: done

File: client/web/src/cody/chat/CodyChatPage.tsx (hunk 2 above)
Comment (thenamankumar) [reply client/web/src/cody/chat/CodyChatPage.tsx:49]: done

File: client/web/src/cody/chat/AskCodyThumbnail.tsx (hunk 1 above)
Comment (thenamankumar) [reply client/web/src/cody/chat/AskCodyThumbnail.tsx]: replaced svg with png

File: client/web/src/cody/chat/AskCodyThumbnail.tsx (hunk 4 above)
Comment (thenamankumar) [reply client/web/src/cody/chat/AskCodyThumbnail.tsx]: replaced svg with png

File: client/web/src/cody/chat/VsCodeExtensionThumbnail.tsx (hunk 5 above)
Comment (thenamankumar) [reply client/web/src/cody/chat/VsCodeExtensionThumbnail.tsx]: replaced svg with png

**Review by almeidapaulooliveira — APPROVED** [review]

**Pull request merged**